src/web/mod.rs

```diff
@@ -269,25 +269,43 @@ impl Handler for CratesfyiHandler {
     }
 }
 
+struct MatchVersion {
+    /// Represents the crate name that was found when attempting to load a crate release.
+    ///
+    /// `match_version` will attempt to match a provided crate name against similar crate names with
+    /// dashes (`-`) replaced with underscores (`_`) and vice versa. If
+    pub corrected_name: Option<String>,
+    pub version: MatchSemver,
+}
+
+impl MatchVersion {
+    /// If the matched version was an exact match to the requested crate name, returns the
+    /// `MatchSemver` for the query. If the lookup required a dash/underscore conversion, returns
+    /// `None`.
+    fn assume_exact(self) -> Option<MatchSemver> {
+        if self.corrected_name.is_none() {
+            Some(self.version)
+        } else {
+            None
+        }
+    }
+}
+
 /// Represents the possible results of attempting to load a version requirement.
-enum MatchVersion {
+enum MatchSemver {
     /// `match_version` was given an exact version, which matched a saved crate version.
     Exact(String),
     /// `match_version` was given a semver version requirement, which matched the given saved crate
     /// version.
     Semver(String),
-    /// `match_version` was given a version requirement which did not match any saved crate
-    /// versions.
-    None,
 }
```

Comment:
Is there a reason you changed all these `MatchVersion`s to `Option<MatchSemver>`? So you can use the methods on `Option`?